### PR TITLE
deriveFixedPoint macro annotation

### PR DESCRIPTION
**Pull request by pepegar**
@andyscott I'm not sure if I need to add something to the bazel build files, since I'm not an user of it

### CODE REVIEW

File: modules/core/src/main/scala/qq/droste/fixie.scala [hunk 1]
```diff
@@ -0,0 +1,189 @@
+package qq.droste
+
+import scala.reflect.macros.blackbox
+import scala.language.experimental.macros
+import scala.annotation.StaticAnnotation
+import scala.annotation.compileTimeOnly
+
+@compileTimeOnly("enable macro paradise to expand macro annotations")
+class fixie extends StaticAnnotation {
```
Comment (andyscott) [inline modules/core/src/main/scala/qq/droste/fixie.scala]: How do you feel about renaming `fixie` to `deriveFixedPoint`?

File: modules/core/src/main/scala/qq/droste/fixie.scala [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+        case c: ClassDef  => convertCaseClass(c)
+        case _            => sys.error("Nope")
+      }
+
+    val nonRecursiveAdt: ClassDef =
+      q"""sealed trait $NonRecursiveAdtName[..${clait.tparams}, $A] extends Product with Serializable"""
+
+    val λ: TypeDef = q"type λ[α] = $NonRecursiveAdtName[..${clait.tparams.map(_.name)}, α]"
+
+    val functorInstance = q"""
```
Comment (andyscott) [inline modules/core/src/main/scala/qq/droste/fixie.scala]: Can we provide a `Traverse` instead?

File: build.sbt [hunk 3]
```diff
@@ -19,7 +19,10 @@ lazy val core = module("core")
   .dependsOn(meta)
   .settings(libraryDependencies ++= Seq(
     "org.typelevel" %%% "cats-core"   % "1.1.0",
-    "org.typelevel" %%% "cats-free"   % "1.1.0"))
+    "org.typelevel" %%% "cats-free"   % "1.1.0",
+    "org.typelevel" %%% "kittens"     % "1.1.0",
```
Comment (andyscott) [inline build.sbt]: This brings in the kittens dependency. Long term we should see about removing this dependency and instead having macros to generate Functor/Traverse instances without Shapeless.

For now, can we split fixie into a separate module to better isolate the additional dependency? I need to keep the core module dependency free with the exception of the binary compat guaranteed cats modules.

For splitting, what do you think of `droste-extras` or another similar name?

**Comment by andyscott** [comment]
Overall, looks good. My only blocking issue is dependency isolation-- see the note.

This opens the door to adding a few other nice macro helpers too :). Very exciting!

File: modules/core/src/main/scala/qq/droste/fixie.scala (hunk 2 above)
Comment (pepegar) [reply modules/core/src/main/scala/qq/droste/fixie.scala]: Currently kittens doesn't derive traverse. Only `Functor` and `Foldable` in that hierarchy

File: build.sbt (hunk 3 above)
Comment (pepegar) [reply build.sbt]: Yeah, makes sense. Maybe `droste-macros` makes more sense? We could put all our macros there and isolate the use of `macro paradise`.

File: modules/core/src/main/scala/qq/droste/fixie.scala (hunk 1 above)
Comment (pepegar) [reply modules/core/src/main/scala/qq/droste/fixie.scala]: Sure thing! Now that it'll be part of droste, the old name doesn't make sense

**Comment by pepegar** [comment]
@andyscott shall we merge this now?

File: build.sbt [hunk 4]
```diff
@@ -19,11 +19,24 @@ lazy val core = module("core")
...
+    compilerPlugin("org.scalamacros" % "paradise" % "2.1.0" cross CrossVersion.patch)
+  ))
 
 lazy val coreJVM = core.jvm
 lazy val coreJS  = core.js
 
+lazy val macros = module("macros")
```
Comment (andyscott) [inline build.sbt]: Let's add the macros project to the aggregate list at the top on the root project

**Comment by pepegar** [comment]
:tada:

**Pull request merged**